src/extension/wallet/receive/lightning-enter-amount.tsx

```diff
@@ -48,135 +45,95 @@ const LightningAmount: React.FC = () => {
     fetchData().catch(console.error);
   }, []);
 
+  useEffect(() => {
+    // get maximal and minimal amount for pair
+    const fetchAmount = async () => {
+      const amount = await receiveFlowRepository.getAmount();
+      if (amount) {
+        setSwapAmount(amount);
+        setTouched(true);
+      }
+    };
+    fetchAmount().catch(console.error);
+  }, []);
+
+  const makeSwap = async (): Promise<SwapData> => {
+    // we will create an ephemeral key pair:
+    // - it will generate a public key to be used with the Boltz swap
+    // - later we will sign the claim transaction with the private key
+    const claimPrivateKey = randomBytes(32);
+    const claimKeyPair = ECPairFactory(ecc).fromPrivateKey(claimPrivateKey);
+    const claimPublicKey = claimKeyPair.publicKey;
```

Comment:
Shall we derive a well-known public key instead? Just thinking what if the user removes/uninstalls marina before completing the swap?

Reply:
I guess we don't loose anything by using it.
Now fixing merge conflicts 😬

Reply:
@tiero in a claim transaction we need the full key pair: public and private key.
If we derive the public key from a marina address, that means we'll need to ask the user for his password in order to have the private key (need to sign the claim transaction).

From a UX perspective, I prefer NOT to ask users for a password for them to RECEIVE.
I don't think is fair for thousands of users the need to insert a password to receive just because one guy could remove/uninstall marina before completing the swap.

What do you think?

Reply:
Also, I guess if you do not claim, the LN payment will simply be refunded to the sender if never claimed on-chain passing the right preimage and the righ signature.

So agreed, let's make it random and store it.